src/ntlm.h

```diff
@@ -128,6 +128,9 @@ struct ntlm_signseal_state {
     bool ext_sec;
 };
 
+#define MAX_SIDS_COUNT 256
+typedef char ntlm_raw_sid[68]; /* Max size of SID is 68 bytes */
```

Comment:
I am not ok with this, it makes a lot of assumptions on what a sid is, how the compilers lays out the source structure and that the sid definition fro the winbind headers will never change.
It also makes it completely opaque for any client requiring them to extract back the data.
If we want something so tightly coupled and opaque, we may as well just return a pointer to th wbcAuthUserInfo structure and hope all goes well (I am not advocating this to be clear).

Reply:
SID binary structure is the same for a long time and cannot just change (Microsoft can increment the version and use new format, but leave struct size unchanged).
Also compiler can't change anything here because struct is well-formed (alignment of each field is good, see wbcDomainSid).
I have seen few implementations of this structure but all of them are binary equal to wbcDomainSid (including MS code). So if application works with SID, I'm pretty sure it will be able to interpret this representation.

Returning more details or even whole wbcAuthUserInfo struct is a good point, but client will need to use winbind client headers to interpret it - this is not always good. Maybe...

Reply:
There are three official SID representations (see 2.4.2 SID of MS-DTYP), the string form, the packet representation form and the RPC marshalled form.

Winbindd uses the packet representation form, so that is what is exposes. It is generally similar to what you see expressed in source code form elsewhere, but it is not a standard per se.

It is also unclear which form would be better for a GSS-NTLMSSP client.
I can see clients that would greatly prefer the string form over a binary opaque structure.

If you really want a binary structure though , I suggest you reproduce it as a structure in this PR and do element assignment to copy the sids over via a copy function.
But I am really tempted by the idea of having an array of strings instead, even if in some cases that could require parsing back the strings into the binary representation.

What application are you planning to use these in? And would the representation used make a difference to you?

Reply:
We need to use the library as a part of proprietary SMB server for NTLM authentication (SMB_SESSION_SETUP verb of SMB2 protocol) - as a addition to MIT library and Kerberos authentication. As a result of authentication, proper ACCESS_TOKEN must be constructed for user to handle permissions checks later. We store all SIDs in binary form so they can be used in native way during access check.

However I don't think that representation really matters here - since it's done once per session, performance is not very critical here; surely we already have all code to convent SIDs to any form (to string and to RPC representation too).